src/coreclr/System.Private.CoreLib/src/System/Diagnostics/StackFrame.CoreCLR.cs

```diff
@@ -50,5 +53,23 @@ private void BuildStackFrame(int skipFrames, bool needFileInfo)
         }
 
         private static bool AppendStackFrameWithoutMethodBase(StringBuilder sb) => false;
+
+        [DllImport(RuntimeHelpers.QCall, EntryPoint = "StackFrame_GetMethodDescFromNativeIP")]
+        private static extern RuntimeMethodHandleInternal GetMethodDescFromNativeIP(IntPtr ip);
+
+        /// <summary>
+        /// Returns the method info instance for the managed code IP address.
+        /// </summary>
+        /// <param name="ip">code address</param>
+        /// <returns>MethodInfo instance or null if IP not found</returns>
+        internal static MethodInfo? GetMethodInfoFromNativeIP(IntPtr ip)
+        {
+            RuntimeMethodHandleInternal method = GetMethodDescFromNativeIP(ip);
+
+            if (method.Value == IntPtr.Zero)
+                return null;
+
+            return RuntimeType.GetMethodBase(null, method) as MethodInfo;
```

Comment:
It would be more appropriate for method to return `MethodBase` so that this cast is not needed. This cast is filtering constructors. If the IP is inside constructor, VSMac is not going to be able to get the method name.